chainer/training/extensions/_snapshot.py

```diff
@@ -1,9 +1,72 @@
+import os
+
+import chainer
 from chainer.serializers import npz
 from chainer.training import extension
 from chainer.training.extensions import snapshot_writers
 from chainer.utils import argument
 
 
+def _find_snapshot_files(fmt, path):
+    # Only prefix and suffix match
+
+    # TODO(kuenishi): currently clean format string such as
+    # "snapshot{.iteration}.npz" can only be parsed, but tricky (or
+    # invalid) formats like "snapshot{{.iteration}}.npz" are hard to
+    # detect and to properly show errors, just ignored or fails so
+    # far.
+    prefix = fmt.split('{')[0]
+    suffix = fmt.split('}')[-1]
+
+    matched_files = (file for file in os.listdir(path)
+                     if file.startswith(prefix) and file.endswith(suffix))
+
+    def _prepend_mtime(f):
+        t = os.stat(os.path.join(path, f)).st_mtime
+        return (t, f)
+
+    return sorted(_prepend_mtime(file) for file in matched_files)
+
+
+def find_latest_snapshot(fmt, path):
+    """Finds the latest snapshots in a directory
+
+    Args:
+        fmt (str): format string to match with file names of
+            existing snapshots, where prefix and suffix are
+            only examined. Also, files' staleness is judged
+            by timestamps. The default is metime.
+        path (str): a directory path to search for snapshot files.
+
+    """
+    snapshot_files = _find_snapshot_files(fmt, path)
+
+    if len(snapshot_files) > 0:
+        _, filename = snapshot_files[-1]
+        return filename
```

Comment:
There's no `return` statement after this `if` clause. Please return `None` explicitly.